allennlp/models/multitask.py

```diff
@@ -0,0 +1,195 @@
+from collections import defaultdict
+import inspect
+from typing import Any, Dict, List, Set
+
+from overrides import overrides
+import torch
+
+from allennlp.data import Vocabulary
+from allennlp.modules import Backbone
+from allennlp.models.model import Model
+from allennlp.models.heads import Head
+from allennlp.nn import InitializerApplicator
+
+
+def get_forward_arguments(module: torch.nn.Module) -> Set[str]:
+    signature = inspect.signature(module.forward)
+    return set([arg for arg in signature.parameters if arg != "self"])
+
+
```

Comment:
Does this model first need to be registered, e.g.:

```
@Model.register("multi_task_model")
```

so that it can be picked up in a config file and then all the subsequent arguments passed underneath that `model`? E.g.:

```
"model": {
  "type": "multi_task_model",
    "backbone": {
    ...
    }
```

Reply:
Hmm, yep, that's an oversight.